cpp/src/copying/contiguous_split.cu

```diff
@@ -195,10 +195,15 @@ __device__ void copy_buffer(uint8_t* __restrict__ dst,
       std::size_t idx = (num_bytes - remainder) / 4;
       uint32_t v = remainder > 0 ? (reinterpret_cast<uint32_t const*>(src)[idx] - value_shift) : 0;
       while (remainder) {
-        uint32_t const next = bit_shift > 0 || remainder > 4
+        // if we're doing a validity copy, do we need to read an extra bitmask word to OR it's
+        // relevant bits in?
+        auto const have_extra_rows =
+          bit_shift > 0 && remainder == 4 ? (num_elements * 32) - num_rows < bit_shift : false;
```

Comment:
This looks equivalent to 
```suggestion
          bit_shift > 0 && remainder == 4 && ((num_elements * 32) - num_rows < bit_shift);
```
which to me seems a bit easier to follow

Reply:
Also, `(num_elements * 32) - num_rows < bit_shift` does not change between iterations. Maybe it can be a const variable declared outside of the loop? Giving it a name would also help readability IMO.